packages/ebsi-authorization-client/src/agent/EBSIAuthorizationClient.ts

```diff
@@ -0,0 +1,418 @@
+          {
+            id: `tnt_authorise_credential`,
+            format: 'jwt_vp',
+            path: '$',
+            path_nested: {
+              id: `tnt_authorise_credential`,
+              format: 'jwt_vc',
+              path: '$.vp.verifiableCredential[0]',
+            },
+          },
+        ]
+      default:
+        throw new Error(`${definitionId} is not supported`)
+    }
+  }
+
+  private async ebsiAuthASDiscoveryMetadataGet(): Promise<GetOIDProviderMetadataResponse> {
+    return await (
+      await fetch(`https://api-pilot.ebsi.eu/authorisation/v4/.well-known/openid-configuration`, {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+        },
+      })
+    ).json()
+  }
+
+  private async ebsiAuthASJwksGet(): Promise<GetOIDProviderJwksResponse | ExceptionResponse> {
+    if (!this.discoveryMetadata) {
+      this.discoveryMetadata = await this.ebsiAuthASDiscoveryMetadataGet()
+    }
+    return await (
+      await fetch(`${this.discoveryMetadata.jwks_uri}`, {
+        method: 'GET',
+        headers: {
+          Accept: 'application/jwk-set+json',
+        },
+      })
+    ).json()
+  }
+
+  private async ebsiAuthPresentationDefinitionGet(args: GetPresentationDefinitionArgs): Promise<GetPresentationDefinitionResponse> {
+    const { scope } = args
+    if (!this.discoveryMetadata) {
+      this.discoveryMetadata = await this.ebsiAuthASDiscoveryMetadataGet()
+    }
+    const ebsiScope = Object.keys(EBSIScope)[Object.values(EBSIScope).indexOf(scope)]
+    return await (
+      await fetch(`${this.discoveryMetadata.presentation_definition_endpoint}?scope=openid%20${ebsiScope}`, {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+        },
+      })
+    ).json()
+  }
+
+  private async ebsiAuthAccessTokenGet(args: EBSIAuthAccessTokenGetArgs, context: IRequiredContext): Promise<GetAccessTokenResponse> {
+    const { credential, definitionId, did, kid, alg } = args
+
+    const metadataResponse = await this.ebsiAuthASDiscoveryMetadataGet()
+    if ('status' in metadataResponse) {
+      throw Error(JSON.stringify(metadataResponse))
+    }
+
+    const vc = await this.createVcJwt(
+      {
+        payloadVc: credential,
+        kid,
+        alg,
+      },
+      context,
+    )
+
+    const vpJwt = await this.createVPJwt(
+      {
+        kid,
+        did,
+        vc,
+        audience: did,
+        domain: this.baseUrl,
+      },
+      context,
+    )
+    let descriptorMap = this.getDescriptorMap(definitionId)
+
+    const id = Object.keys(ScopeByDefinition)[Object.values(ScopeByDefinition).indexOf(definitionId)]
+    const presentationSubmission = {
+      id: uuid(),
+      definition_id: id,
+      descriptor_map: descriptorMap,
+    }
+
+    const tokenResponse = await this.getAccessToken({
+      grant_type: 'vp_token',
+      vp_token: vpJwt,
+      presentation_submission: presentationSubmission,
+      scope: EBSIScope.didr_invite,
```

Comment:
Should this `scope` come from args instead of being hardcoded?

Reply:
Retrieving the values from args now